### PR TITLE
cleanup: Remove `Spanned` where possible

### PR DESCRIPTION
**Pull request by petrochenkov**
It generally only makes sense on enums, otherwise it's more convenient to "flatten" it by adding a span field to the struct it wraps.

### CODE REVIEW

File: src/libsyntax/ast.rs [hunk 1]
```diff
@@ -1287,19 +1286,18 @@ pub enum Movability {
...
-
 /// Represents a macro invocation. The `Path` indicates which macro
 /// is being invoked, and the vector of token-trees contains the source
 /// of the macro invocation.
 ///
 /// N.B., the additional ident for a `macro_rules`-style macro is actually
 /// stored in the enclosing item.
 #[derive(Clone, RustcEncodable, RustcDecodable, Debug)]
-pub struct Mac_ {
+pub struct Mac {
```
Comment (eddyb) [inline src/libsyntax/ast.rs]: By this point we can probably rename this to `MacroInvocation` or `MacroCall`.
(but it doesn't need to be in this PR)

**Review by eddyb — APPROVED** [review]

**Comment by Centril** [comment]
Nice change.

...But the PR builder seems unhappy? @bors r-

**Comment by petrochenkov** [comment]
@bors r=eddyb
(Rustdoc updated.)

**Pull request merged**